changelog_unreleased/html/pr-7865.md

```diff
@@ -0,0 +1,32 @@
+#### Prefix-based multiline formatting for the `class` attribute ([#7865](https://github.com/prettier/prettier/pull/7865) by [@thorn0](https://github.com/thorn0))
+
+If class names don't fit within the print width, Prettier now splits them onto multiple lines trying to keep consecutive classes with the same prefix on one line.
+
+<!-- prettier-ignore -->
+```html
+<!-- Input -->
+<div class="ProviderMeasuresContainer__heading-row
+  d-flex flex-column flex-lg-row
+  justify-content-start
+  justify-content-lg-between align-items-start align-items-lg-center">Foo</div>
+
+<!-- Prettier stable -->
+<div
+  class="ProviderMeasuresContainer__heading-row d-flex flex-column flex-lg-row justify-content-start justify-content-lg-between align-items-start align-items-lg-center"
+>
+  Foo
+</div>
+
+<!-- Prettier master -->
+<div
+  class="
+    ProviderMeasuresContainer__heading-row
+    d-flex
+    flex-column flex-lg-row
+    justify-content-start justify-content-lg-between
+    align-items-start align-items-lg-center
+  "
```

Comment:
I'm afraid we will get issues with requests to format each class on each line/don't touch, maybe we put information why we prefer it to docs?

Reply:
> I'm afraid we will get issues with requests to format each class on each line/don't touch, maybe we put information why we prefer it to docs?

For me I prefer one name per line, also consider the following example:

```html
<div class="custom-btn my-very-v-button--mini site-btn my-very-v-button--primary my-very-v-button--ghost">my button</div>
```

After running prettier, I got:

```html
<div
  class="
    custom-btn
    my-very-v-button--mini
    site-btn
    my-very-v-button--primary my-very-v-button--ghost
  "
>
  my button
</div>
```

Given `80` as print width, the 3 names starting with `my-very-v-button` are actually able to fit into one line, but they are not, also it looks better to at least put them next to each other since they share the same prefix, i.e. switch the `site-btn` line with the last line.

So, my suggestion is let's don't group names with same prefix into one line, just let each name stands in it's own line(which is good for Git diff also), anyway an element or component that has more than 10 class names is really edge case, in that case I should consider my way of using class names, there must be some parts that can be improved to avoid too many class names on the same element.

Reply:
@maogongzi 

> the 3 names starting with my-very-v-button are actually able to fit into one line, but they are not

Right because Prettier never reorders anything. However, one of the points of the current proposal is that in real projects, classes usually are already sorted by prefix, and when they are not, it's for a reason.

Reply:
@evilebottnawi 

> I'm afraid we will get issues with requests to format each class on each line/don't touch, maybe we put information why we prefer it to docs?

I added an explanation to the changelog entry. Do you think we also need to write something in the docs about this?